app/src/lib/stores/helpers/branch-pruner.ts

```diff
@@ -87,37 +87,93 @@ export class BranchPruner {
         )
   }
 
-  private async pruneLocalBranches(): Promise<void> {
-    if (this.repository.gitHubRepository === null) {
-      return
+  /**
+   * Locally prune branches when the following criteria are met
+   * 1. deleted on the remote (github.com) and
+   * 2. merged into the repository's default branch and
+   * 3. hasn't been checked out locally since `timeSinceLastCheckout`
+   *
+   * Note: This is ran automatically by calling `start` on the `BranchPruner` instance
+   *
+   * @param timeSinceLastCheckout limits pruning to branches that haven't been checked out since this date (defaults to 2 weeks before today), passing null ignores constraint `3`
+   * @returns true when branches have been prune
+   */
+  public async prune(
+    timeSinceLastCheckout: Date | null = moment()
```

Comment:
I think this is the only part that confuses me about the current changes, because this function can now accept:

 - a `Date` indicating the time since last checkout (not currently used)
 - `null` to ignore the constraint (added to support dev mode testing)
 - `undefined`, which then uses the default `Date` value of `moment().subtract(2, 'weeks').toDate()` (the default behaviour)

Are we able to tighten this up so the API either supports `Date` or `null`? I think this means changing line 237 to this:

```diff
diff --git a/app/src/lib/stores/helpers/branch-pruner.ts b/app/src/lib/stores/helpers/branch-pruner.ts
index 784e8bdd2..167c3f444 100644
--- a/app/src/lib/stores/helpers/branch-pruner.ts
+++ b/app/src/lib/stores/helpers/branch-pruner.ts
@@ -229,7 +229,11 @@ export class BranchPruner {
       return
     }
 
-    const didPruneHappen = await this.prune()
+    const timeSinceLastCheckout = moment()
+      .subtract(2, 'weeks')
+      .toDate()
+
+    const didPruneHappen = await this.prune(timeSinceLastCheckout)
     await this.repositoriesStore.updateLastPruneAttemptDate(
       this.repository,
       Date.now()
```